pennylane_lightning/core/src/simulators/lightning_kokkos/gates/tests/Test_StateVectorKokkos_Generator.cpp

```diff
@@ -695,6 +708,7 @@ TEMPLATE_TEST_CASE("StateVectorKokkosManaged::applyGeneratorCRZ",
 
 TEMPLATE_TEST_CASE("StateVectorKokkosManaged::applyGeneratorMultiRZ",
                    "[StateVectorKokkosManaged_Generator]", float, double) {
+    const bool inverse = GENERATE(true, false);
```

Comment:
What is the `GENERATE` call for? I assume since the next section is scoped it creates two test variants at this point?

Reply:
`GENERATE` is akin to `pytest.mark.parametrize`. Here is a [link](https://github.com/catchorg/Catch2/blob/devel/docs/generators.md) to the doc. Most gate functors are templated on `inverse`, but the `inverse=true` branches are almost never tested as pointed by Codecov. For example, in the previous `multiQubitOpFunctor` implementation, the [inverse branch](https://github.com/PennyLaneAI/pennylane-lightning-kokkos/pull/75/files#diff-f158dd60dba477dda099bef0bf80cdd6d7606d3e66aa035ddf294470e6618dfbL228) had `const std::size_t n_wires = dim;` instead of `const std::size_t n_wires = wires.size();` (the former is `2**wires.size()` and the latter is correct). I thus introduced parametrization over `inverse` to cover all `inverse` branches, as much as possible.

Reply:
The use of Catch2 data generators is a nice addition to our test stack.